### ISSUE TITLE
xgb.cv with early stopping returns wrong best iteration when multiple metrics are used

### ISSUE THREAD

**Issue by goerlitz** [dmlc/xgboost#4630]
I've been using xgb.cv with early stopping to determine the best number of training rounds. The [documentation for early-stopping](https://xgboost.readthedocs.io/en/latest/python/python_intro.html#early-stopping) only mentions `xgb.train` and `param['eval_metric']` and says that 

> if you specify more than one evaluation metric the last one in param['eval_metric'] is used for early stopping

I assumed that the same would be true for `xgb.cv` and the parameter `metrics`. However, when using multiple metrics it does not return the correct number for the best iteration.

Following minimal code example demonstrates the encountered problem:

```
import sklearn.datasets as datasets
import xgboost as xgb

print(xgb.__version__)

X, y = datasets.load_breast_cancer(return_X_y=True)
train = xgb.DMatrix(X, y)

params = {'objective':'binary:logistic'}

metrics = [['auc'], ['error'], ['logloss'], ['logloss', 'auc'], ['logloss', 'error'], ['error', 'logloss']]
for i, m in enumerate(metrics):
    result = xgb.cv(params, train, 1000, 5, stratified=True, metrics=m, early_stopping_rounds=50, seed=42)
    print(f'\n{i+1}: metrics = {m}\n', result.tail(1)[[f"test-{x}-mean" for x in m]])
```
Output:

```
0.82

1: metrics = ['auc']
      test-auc-mean
170       0.993797

2: metrics = ['error']
     test-error-mean
69         0.031612

3: metrics = ['logloss']
     test-logloss-mean
64           0.096297

4: metrics = ['logloss', 'auc']
     test-logloss-mean  test-auc-mean
64           0.096297       0.993461

5: metrics = ['logloss', 'error']
     test-logloss-mean  test-error-mean
64           0.096297         0.033382

6: metrics = ['error', 'logloss']
     test-error-mean  test-logloss-mean
64         0.033382           0.096297
```

In cases 1-3 with a single metric we get different values for the best number of iteration, i.e. 170, 69, 64, respectively. Cases 4-6 have two metrics where the second metric is the same as in cases 1-3. Therefore, the expected values for the best number of iterations should be the same as in 1-3. Instead, it's always 64 - which corresponds to the best number of iterations of the 'logloss' metric.

**Comment by hcho3** [comment]
@goerlitz It turns out that there was a bug that caused the metrics to be re-ordered. #4638 fixes the bug, by preserving the original ordering of validation sets and evaluation metrics.

Also, #4638 clarifies the API doc to describe how early stopping works when multiple validation sets and metrics are given: Always, the last validation set and the last metric will be used.
Preview: https://hcho3-xgboost.readthedocs.io/en/fix_early_stopping_python/python/python_api.html#xgboost.train
